src/App.tsx

```diff
@@ -2,109 +1,63 @@
+import React, { useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
 import './App.css';
-import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+import Pagination from './components/Pagination';
+import FormGroup from './components/FormGroup';
+import handleCurrentPage from './handlers/handleCurrentPage';
+import handleCurrentOption from './handlers/handleOptionChange';
 
 export const App: React.FC = () => {
+  const total = import.meta.env.VITE_PAGES_TOTAL;
+  const [perPage, setPerPage] = useState<number>(5);
+  const [currentPage, setCurrentPage] = useState<number>(
+    +import.meta.env.VITE_START_PAGE,
+  );
+  const [searchParams, setSearchParams] = useSearchParams();
+  const pageItemsCount = Math.ceil(+total / perPage);
```

Comment:
The calculation of `pageItemsCount` might lead to unexpected results if `total` is not a valid number. Ensure that `total` is a valid number to avoid potential issues.